From 4749b0c06f1d3c5f4b2a1f73325c53ff5e50a841 Mon Sep 17 00:00:00 2001
From: Sachin Prabhu <sprabhu@redhat.com>
Date: Mon, 28 Jul 2014 12:16:46 -0400
Subject: [fs] cifs: remove unused variable

Message-id: <1406549822-10376-54-git-send-email-sprabhu@redhat.com>
Patchwork-id: 86795
O-Subject: [RHEL7.1 PATCH 53/69] [bz1082049] fs: cifs: remove unused variable.
Bugzilla: 1082049
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Cyril Roelandt <tipecaml@gmail.com>

In SMB2_set_compression(), the "res_key" variable is only initialized to NULL
and later kfreed. It is therefore useless and should be removed.

Found with the following semantic patch:

<smpl>
@@
identifier foo;
identifier f;
type T;
@@
* f(...) {
...
* T *foo = NULL;
... when forall
    when != foo
* kfree(foo);
...
}
</smpl>

Signed-off-by: Cyril Roelandt <tipecaml@gmail.com>
Signed-off-by: Steve French <sfrench@us.ibm.com>
(cherry picked from commit 8e3ecc87695f4a7e9e217ebd55ca6a39b6a451b8)

Signed-off-by: Sachin Prabhu <sprabhu@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/cifs/smb2pdu.c b/fs/cifs/smb2pdu.c
index 8603447..3802f8c 100644
--- a/fs/cifs/smb2pdu.c
+++ b/fs/cifs/smb2pdu.c
@@ -1352,7 +1352,6 @@ SMB2_set_compression(const unsigned int xid, struct cifs_tcon *tcon,
        u64 persistent_fid, u64 volatile_fid)
 {
  int rc;
- char *res_key = NULL;
  struct  compress_ioctl fsctl_input;
  char *ret_data = NULL;
 
@@ -1365,7 +1364,6 @@ SMB2_set_compression(const unsigned int xid, struct cifs_tcon *tcon,
    2 /* in data len */, &ret_data /* out data */, NULL);
 
  cifs_dbg(FYI, "set compression rc %d\n", rc);
- kfree(res_key);
 
  return rc;
 }
-- 
1.7.1